python-package/lightgbm/dask.py

```diff
@@ -244,6 +244,7 @@ def _train(client, data, label, params, model_factory, sample_weight=None, group
     for tree_learner_param in _ConfigAliases.get('tree_learner'):
         tree_learner = params.get(tree_learner_param)
         if tree_learner is not None:
+            params['tree_learner'] = tree_learner
```

Comment:
I know this is a little awkward in the case where aliases are used, because multiple aliases for tree_learner might exist in params.

That change will only live on `master` for a short time, because I'm cleaning it up in https://github.com/microsoft/LightGBM/pull/3813#discussion_r562754584 right now